pkg/client/client.go

```diff
@@ -638,6 +643,87 @@ func newConfiguration(ctrlRuntimeClient client.Client, secretName, namespace str
 	return config, nil
 }
 
+func fetchCredentialsIniFromSecret(secret *corev1.Secret, config *providers.Configuration) error {
+	creds, ok := secret.Data["credentials"]
+	if !ok {
+		return fmt.Errorf("failed to fetch key 'credentials' in secret data")
+	}
+	inifile, err := ini.Load(creds)
+	if err != nil {
+		return err
+	}
+	s, err := inifile.GetSection(kubeIniDefaultSection)
+	if err != nil {
+		return err
+	}
+	if s.HasKey(kubeCredentialTypeString) {
```

Comment:
I would reverse this condition to minimise indentation

```
if !s.HasKey(...) {
  // Skip this one because...
  return nil
}

accessType, err := s.GetKey(...)
...
```